docs/drafts/version-ranges.md

```diff
@@ -0,0 +1,93 @@
+# Operator version ranges
+
+This document explains how to specify a version range to install or update an Operator with OLM 1.0.
+
+You define a version range in an Operator's custom resource (CR) file.
+
+## Specifying a version range in the CR
```

Comment:
I removed the other sections on the `spec.version` field. They didn't make sense with the expanded content, and I think it will take a bit more work to incorporate the content into one doc on the `spec.version` field. I created a follow up issue here: https://github.com/operator-framework/operator-controller/issues/551